lib/mocks/billdispenser.js

```diff
@@ -40,32 +48,47 @@ BillDispenser.prototype.init = function init (data) {
   })
 }
 
-BillDispenser.prototype.dispense = function dispense (notes, cb) {
+BillDispenser.prototype.dispense = function dispense (notes) {
   var self = this
   console.log('Mock dispensing...', notes, self.cassettes)
 
-  return new Promise((resolve, reject) => {
-    setTimeout(function () {
+  return Bluebird.delay(2000)
+
+    // check if notes is matching any error
+    .then(() => {
+      const error = this.matchMockedErrors(notes)
+
+      if (!_.isUndefined(error)) {
+        const dispenseErr = new Error(parseInt(error, 16))
+        dispenseErr.name = 'DispenserError'
+
+        throw dispenseErr
+      }
+    })
+
+    // check if balance is available
+    .then(() => {
       const count = _.min([_.size(notes), _.size(self.cassettes)])
       for (var i = 0; i < count; i++) {
         if (notes[i] > self.cassettes[i].count) {
-          var err = new Error('Mocked BillDispenser: not enough cash')
-          err.statusCode = 570
-          if (cb) cb(err)
-          reject(err)
-          return false
+          var dispenseErr = new Error('not enough cash')
```

Comment:
If you're making a lot of changes in a file, it's worth changing all the `var`s to `const`s.